.github/actions/run-in-docker-action/action.yml

```diff
@@ -0,0 +1,40 @@
+name: 'CI script in Docker'
+description: 'Run a CI script in a Docker container'
+inputs:
+  dockerfile:
+    description: 'A Dockerfile that defines an image'
+    required: true
+  tag:
+    description: 'A tag of an image'
+    required: true
+  command:
+    description: 'A command to run in a container'
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - run: |
+        for var in "$(echo '${{ toJSON(matrix.configuration.env_vars) }}' | jq -r 'to_entries | .[] | "\(.key)=\(.value)"')"; do
+          echo "$var" >> "$GITHUB_ENV"
+        done
+        echo "MAKEFLAGS=-j$(($(nproc) + 1))" >> "$GITHUB_ENV"
```

Comment:
nit: This will probably override `MAKEFLAGS` if in `env_vars`.  Perhaps it's a bit cleaner to simply *prepend* `MAKEFLAGS` with `$(($(nproc) + 1))`, and do this inside `ci.sh`?

Reply:
To keep the scope of this PR limited, is it OK to just set `mingw_debian.env.MAKEFLAGS: '-j3'`?

Reply:
> Or rely on the global `env.MAKEFLAGS: '-j4'`?

[Implemented](https://github.com/bitcoin-core/secp256k1/pull/1398#issuecomment-1675222830).